packages/salesforcedx-vscode-apex/src/codecoverage/colorizer.ts

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) 2019, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as fs from 'fs';
+import * as path from 'path';
+import { Range, TextDocument, TextEditor, window, workspace } from 'vscode';
+import {
+  coveredLinesDecorationType,
+  uncoveredLinesDecorationType
+} from './decorations';
+import { StatusBarToggle } from './statusBarToggle';
+
+const apexDirPath = path.join(
+  workspace!.workspaceFolders![0].uri.fsPath,
+  '.sfdx',
+  'tools',
+  'testresults',
+  'apex'
+);
+
+export function getLineRange(
+  document: TextDocument,
+  lineNumber: number
+): Range {
+  const adjustedLineNumber = lineNumber - 1;
+  const firstLine = document.lineAt(adjustedLineNumber);
+
+  return new Range(
+    adjustedLineNumber,
+    firstLine.range.start.character,
+    adjustedLineNumber,
+    firstLine.range.end.character
+  );
+}
+
+export type CoverageTestResult = {
+  coverage: {
+    coverage: CoverageItem[];
+  };
+};
+
+export type CoverageItem = {
+  id: string;
+  name: string;
+  totalLines: number;
+  lines: { [key: string]: number };
+};
+
+function getTestRunId() {
+  const testRunIdFile = path.join(apexDirPath, 'test-run-id.txt');
+  if (!fs.existsSync(testRunIdFile)) {
+    throw new Error('No test run information was found for this project.');
+  }
+  return fs.readFileSync(testRunIdFile, 'utf8');
+}
+
+function getCoverageData() {
+  const testRunId = getTestRunId();
+  const testResultFilePath = path.join(
+    apexDirPath,
+    `test-result-${testRunId}.json`
+  );
+
+  if (!fs.existsSync(testResultFilePath)) {
+    throw new Error(
+      `No code coverage information was found for test run ${testRunId}`
+    );
+  }
+  const testResultOutput = fs.readFileSync(testResultFilePath, 'utf8');
+  const codeCoverage = JSON.parse(testResultOutput) as CoverageTestResult;
+  if (codeCoverage.coverage === undefined) {
+    throw new Error(
+      `No code coverage information was found for test run ${testRunId}.`
```

Comment:
Error showed when the test results do not include code coverage information.

Reply:
What would we recommend that people do next? Set `"salesforcedx-vscode-core.retrieve-test-code-coverage": true`? Or did something else probably cause this error?

Reply:
Yes, running tests without that setting on.